### PR TITLE
Move token related actions into separate controller and add tests

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
[Move token related action into own controller](https://trello.com/c/gcKZnuOy/719-move-token-related-action-into-own-controller)

File: src/api/app/controllers/person/token_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    end
+
+    # POST /person/<login>/token
+    def create
+      authorize @user, :update?
+
+      pkg = nil
+      if params[:project] || params[:package]
+        pkg = Package.get_by_project_and_name(params[:project], params[:package])
+      end
```
Comment (ChrisBr) [inline src/api/app/controllers/person/token_controller.rb]: Maybe move it to a function and then you could do:
``
@token = Token::Service.create(user: @user, package: set_package)
``

File: src/api/app/controllers/person/token_controller.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      @token = Token::Service.create(user: @user, package: pkg)
+    end
+
+    # DELETE /person/<login>/token/<id>
+    def delete
+      authorize @user, :update?
+
+      token = Token::Service.where(user_id: @user.id, id: params[:id]).first
```
Comment (ChrisBr) [inline src/api/app/controllers/person/token_controller.rb]: ``find_by(...)``?

File: src/api/app/controllers/person/token_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+require 'xmlhash'
+module Person
+  class TokenController < ApplicationController
+    before_action :set_user, only: [:index, :create, :delete]
```
Comment (hennevogel) [inline src/api/app/controllers/person/token_controller.rb]: :thought_balloon: are there any other actions?

File: src/api/app/controllers/person/token_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+      @list = @user.service_tokens
+    end
+
+    # POST /person/<login>/token
+    def create
+      authorize @user, :update?
+
+      pkg = nil
+      if params[:project] || params[:package]
```
Comment (hennevogel) [inline src/api/app/controllers/person/token_controller.rb]: What if there is `:project` but not `:package`?

File: src/api/app/controllers/person/token_controller.rb (hunk 2 above)
Comment (hennevogel) [reply src/api/app/controllers/person/token_controller.rb]: `@user.service_tokens.find!(params[:id])`

then you can skip `render_error` below right?

File: src/api/spec/routing/api_matcher_spec.rb [hunk 5]
```diff
@@ -3,6 +3,15 @@
 RSpec.describe 'APIMatcher' do
   it { expect(get('/distributions?format=xml')).to route_to(controller: 'distributions', action: 'index', format: 'xml') }
 
+  context 'old person token routes get properly directed' do
```
Comment (hennevogel) [inline src/api/spec/routing/api_matcher_spec.rb]: Hm, why do you test this?

File: src/api/spec/controllers/person/token_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,105 @@
+require 'rails_helper'
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. Some of the actions
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
```
Comment (hennevogel) [inline src/api/spec/controllers/person/token_controller_spec.rb]: Why do you have those comments if there are no backend calls that provide meaningful response (a.k.a. `vcr: false`)?

File: src/api/app/controllers/person/token_controller.rb (hunk 3 above)
Comment (bgeuken) [reply src/api/app/controllers/person/token_controller.rb]: :blush: no

File: src/api/app/controllers/person/token_controller.rb (hunk 4 above)
Comment (bgeuken) [reply src/api/app/controllers/person/token_controller.rb]: I think it's a bug, but I didn't want to change it altogether. I also have a commit which handles invalid parameters, but want to have this merged first.

File: src/api/spec/routing/api_matcher_spec.rb (hunk 5 above)
Comment (bgeuken) [reply src/api/spec/routing/api_matcher_spec.rb]: To be sure that I don't break this when changing the routes.

File: src/api/spec/controllers/person/token_controller_spec.rb (hunk 6 above)
Comment (bgeuken) [reply src/api/spec/controllers/person/token_controller_spec.rb]: Right, I shall drop them. I just c&p'ed this from the preson controller spec

File: src/api/spec/routing/api_matcher_spec.rb (hunk 5 above)
Comment (bgeuken) [reply src/api/spec/routing/api_matcher_spec.rb]: Initially I wanted to use ` resources ` in the routes, but I dropped that. By now the new route is tested. So I dropped the test now.

**Comment by bgeuken** [comment]
@hennevogel updated

File: src/api/app/controllers/person/token_controller.rb (hunk 4 above)
Comment (hennevogel) [reply src/api/app/controllers/person/token_controller.rb]: k

**Pull request merged**